test/tvl-staking/GaugeController.ts

```diff
@@ -0,0 +1,281 @@
+/* eslint-disable @typescript-eslint/dot-notation */
+/* eslint-disable no-await-in-loop */
+/* eslint-disable @typescript-eslint/naming-convention */
+
+import { Contract, ContractFactory } from "@ethersproject/contracts";
+import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
+import chai, { expect } from "chai";
+import { ethers } from "hardhat";
+import { solidity } from "ethereum-waffle";
+import { getTimestamp, takeSnapshot, revertToSnapShot } from "../utils/time";
+import { BN, simpleToExactAmount } from "../utils/math";
+import { StandardAccounts } from "../utils/machines";
+import { Account } from "../../types";
+import { addSnapshotBeforeRestoreAfterEach } from "../common";
+import {
+  ONE_WEEK,
+  ONE_YEAR,
+  DEFAULT_DECIMALS,
+  ZERO_ADDRESS,
+  DEAD_ADDRESS,
+} from "../utils/constants";
+chai.use(solidity);
+
```

Comment:
only added testing on what I changed (disperse_funds).

crv team [testing](https://github.com/curvefi/curve-dao-contracts/tree/master/tests/integration/GaugeController) of gauge controller